src/sentry/static/sentry/app/components/forms/selectControl.tsx

```diff
@@ -48,13 +57,36 @@ const combineStyles = (newStyles, baseStyles) => {
   }, baseStyles);
 };
 
-const SelectControl = props => {
+type ControlProps = React.ComponentProps<typeof ReactSelect> & {
+  theme: Theme;
+  /**
+   * Ref forwarded into ReactSelect component.
+   * The any is inherited from react-select.
+   */
+  forwardedRef: React.Ref<ReactSelect>;
+  /**
+   * Set to true to prefix selected values with content
+   */
+  inFieldLabel?: string;
+  /**
+   * Backwards compatible shim to work with select2 style choice type.
+   */
+  choices?: Choices | ((props: ControlProps) => Choices);
+  /**
+   * Use react-select v2. Deprecated, don't make more of this.
+   */
+  deprecatedSelectControl?: boolean;
+};
+
+type LegacyProps = React.ComponentProps<typeof SelectControlLegacy>;
+
+function SelectControl(props: ControlProps) {
   // TODO(epurkhiser): We should remove all SelectControls (and SelectFields,
   // SelectAsyncFields, etc) that are using this prop, before we can remove the
   // v1 react-select component.
   if (props.deprecatedSelectControl) {
     const {deprecatedSelectControl: _, ...legacyProps} = props;
-    return <SelectControlLegacy {...legacyProps} />;
+    return <SelectControlLegacy {...((legacyProps as unknown) as LegacyProps)} />;
```

Comment:
In parallel I'm trying to make this path go away. I didn't want to put the effort into making types flow nicely here.